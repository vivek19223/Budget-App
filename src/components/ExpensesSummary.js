import React from 'react';
import ExpenseTotal from '../selectors/expense-total';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral'

export const ExpensesSummary = props => {
    const total = numeral(ExpenseTotal (props.expenses)).format('0,0')
    return (
        <div>
            {props.expenses.length > 1
            ? <h1>Viewing {props.expenses.length} expenses totalling {total}</h1>
            : props.expenses.length === 1 &&
             <h1>Viewing {props.expenses.length} expense totalling {total} </h1>}
        </div>
    )
};

const mapStateToProps = state => {
  return {
    expenses: selectExpenses (state.expenses, state.filters),
  };
};
export default connect (mapStateToProps) (ExpensesSummary);
