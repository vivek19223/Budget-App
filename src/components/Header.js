import React from 'react'
import {   NavLink } from 'react-router-dom';
import { startLogout, startLogin} from '../actions/auth';
import { connect } from 'react-redux';

export const Header = ({ startLogout })=>(
    <header>
        <h1>Expensify</h1>
        <div>
        <NavLink to="/dashboard" activeClassName="is-active" exact={ true }>Dashboard</NavLink><br />
        <NavLink to = "/create" activeClassName="is-active">Create Expense</NavLink><br />
        <button onClick= { startLogout }>Logout</button>
        </div>
    </header>
    );

const mapDispatchToProps = (dispatch) => ({
    startLogout : ()  => dispatch(startLogout())
})

export default connect(undefined,mapDispatchToProps)(Header);