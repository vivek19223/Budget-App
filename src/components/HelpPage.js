import React from 'react';

const HelpPage = () => (
    <div>
      This is a help component.
    </div>
  );

export default HelpPage;