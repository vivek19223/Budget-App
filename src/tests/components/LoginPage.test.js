import React from 'react'
import { shallow } from 'enzyme'
import { LoginPage } from '../../components/LoginPage'

test('Should test login page correctly',()=>{
    const wrapper = shallow(<LoginPage startLogin={ ()=> { }}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Should call startLogin on button click',()=>{
    const onClickSpy = jest.fn()
    const wrapper = shallow(<LoginPage startLogin={ onClickSpy}/>)
    wrapper.find('button').simulate('click')
    expect(onClickSpy).toHaveBeenCalled()
})