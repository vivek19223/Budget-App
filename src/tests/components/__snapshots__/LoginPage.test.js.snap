// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test login page correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Login 
  </button>
</div>
`;
